analytics/msf4j-analytics/src/main/java/org/wso2/msf4j/analytics/httpmonitoring/config/HTTPMonitoringConfigBuilder.java

```diff
@@ -15,29 +15,61 @@
  */
 package org.wso2.msf4j.analytics.httpmonitoring.config;
 
-import org.wso2.carbon.metrics.core.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.config.ConfigProviderFactory;
+import org.wso2.carbon.config.ConfigurationException;
+import org.wso2.carbon.config.provider.ConfigProvider;
 import org.wso2.msf4j.analytics.httpmonitoring.config.model.HTTPMonitoringConfig;
-import org.yaml.snakeyaml.Yaml;
+import org.wso2.msf4j.analytics.internal.DataHolder;
 
-import java.util.Optional;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 
 /**
  * Build {@link HTTPMonitoringConfig} from the YAML file
  */
 public final class HTTPMonitoringConfigBuilder {
 
+    private static final Logger logger = LoggerFactory.getLogger(HTTPMonitoringConfigBuilder.class);
+    private static final String DEPLOYMENT_YAML_SYS_PROPERTY = "msf4j.conf";
+
     public static HTTPMonitoringConfig build() {
-        Optional<String> metricsConfigFileContent = Utils.readFile("http-monitoring.conf", "http-monitoring.yml");
-        if (metricsConfigFileContent.isPresent()) {
+        ConfigProvider configProvider = DataHolder.getInstance().getConfigProvider();
+        HTTPMonitoringConfig configurationObject = null;
+        if (configProvider == null) {
+            if (DataHolder.getInstance().getBundleContext() != null) {
+                throw new RuntimeException(
+                        "Failed to populate HTTPMonitoringConfig Configuration. Config Provider is Null.");
+            }
+            //Standalone mode
+            String deploymentYamlPath = System.getProperty(DEPLOYMENT_YAML_SYS_PROPERTY);
+            if (deploymentYamlPath == null || deploymentYamlPath.isEmpty()) {
+                logger.info("System property '" + DEPLOYMENT_YAML_SYS_PROPERTY +
+                            "' is not set. Default deployment.yaml file will be used.");
+                deploymentYamlPath = HTTPMonitoringConfig.class.getResource("/deployment.yaml").getPath();
+            } else if (!Files.exists(Paths.get(deploymentYamlPath))) {
+                String msg = "Couldn't find " + deploymentYamlPath;
+                throw new RuntimeException(msg);
```

Comment:
We can reduce above two lines into one by straight-away concatenating above message as a param to RuntimeException, WDYT ? Since there is no other reuse of msg property.